cmds/legacy_boot/legacy_boot.go

```diff
@@ -0,0 +1,383 @@
+			}
+			if (strings.HasPrefix(trimmed_line, "menuentry ") == true) && (status == 0) {
+				status = 1
+				return_value = append(return_value, trimmed_line)
+			}
+			if (strings.HasPrefix(trimmed_line, "linux ") == true) && (status == 1) {
+				status = 2
+				return_value = append(return_value, trimmed_line)
+			}
+			if (strings.HasPrefix(trimmed_line, "initrd ") == true) && (status == 2) {
+				status = 0
+				return_value = append(return_value, trimmed_line)
+			}
+		}
+	}
+	return return_value, int_return
+
+}
+
+func copy_local(path string) string {
+	var dest string
+	result := strings.Split(path, "/")
+	for _, entry := range result {
+		dest = entry
+	}
+	dest = "/tmp/" + dest
+	srcFile, err := os.Open(path)
+	check(err)
+	defer srcFile.Close()
+
+	destFile, err := os.Create(dest) // creates if file doesn't exist
+	check(err)
+	defer destFile.Close()
+
+	_, err = io.Copy(destFile, srcFile) // check first var for number of bytes copied
+	check(err)
+
+	err = destFile.Sync()
+	check(err)
+	return dest
+}
+
+// This function is booting new kernel based on the content of grub.cfg
+
+func kexec_entry(grub_conf_path string, mount_point string) {
+	var file_menu_content []string
+	var entry int
+	var local_kernel_path string
+	var local_initrd_path string
+	if verbose {
+		println(grub_conf_path)
+	}
+	file_menu_content, entry = get_file_menu_content(grub_conf_path)
+	var kernel string
+	var kernel_parameter string
+	var initrd string
+	var kernel_infos []string
+	kernel_infos = strings.Fields(file_menu_content[3*entry+1])
+	kernel = kernel_infos[1]
+	var count int
+	count = 0
+	for _, field := range kernel_infos {
+		if count > 1 {
+			kernel_parameter = kernel_parameter + " " + field
+		}
+		count = count + 1
+	}
+	fmt.Sscanf(file_menu_content[3*entry+2], "initrd %s", &initrd)
+	if verbose {
+		println("************** boot parameters  ********************")
+		println(kernel)
+		println(kernel_parameter)
+		println(initrd)
+		println("****************************************************")
+	}
+	local_kernel_path = copy_local(mount_point + kernel)
+	local_initrd_path = copy_local(mount_point + initrd)
+	if verbose {
+		println(local_kernel_path)
+	}
+	umount_entry(mount_point)
+	// We can kexec the kernel with local_kernel_path as kernel entry, kernel_parameter as parameter and initrd as initrd !
+	log.Printf("Loading %s for kernel\n", local_kernel_path)
+
+	kernel_desc, err := os.OpenFile(local_kernel_path, os.O_RDONLY, 0)
+	if err != nil {
+		log.Fatalf("open(%q): %v", local_kernel_path, err)
+	}
+	defer kernel_desc.Close()
+
+	var ramfs *os.File
+	ramfs, err = os.OpenFile(local_initrd_path, os.O_RDONLY, 0)
+	if err != nil {
+		log.Fatalf("open(%q): %v", local_initrd_path, err)
+	}
+	defer ramfs.Close()
+
+	if err := kexec.FileLoad(kernel_desc, ramfs, kernel_parameter); err != nil {
+		log.Fatalf("%v", err)
+	}
+	if err := kexec.Reboot(); err != nil {
+		log.Fatalf("%v", err)
+	}
+
+}
+
+func usage() {
```

Comment:
usage(), is not currently called from anywhere?

Reply:
Thanks. Just removed it from the code.